### PR TITLE
Feature/sph0645 mems mic

### PR DESCRIPTION
**Pull request by MCLEANS**
## Description

1. Added functionality to read audio data from SPH0645 Mic.
2. Convert raw audio data to dB.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: airrohr-firmware/SPH0645.h [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+  I2SC |= I2SRXS;
+}
+
+/**
+ * Set I2S clock.
+ * I2S bits mode only has space for 15 extra bits,
+ * 31 in total. The
+ */
+void
+i2s_set_rate(uint32_t rate)
```
Comment (chegejames) [inline airrohr-firmware/SPH0645.h]: @MCLEANS let's stick to one styling guide.

File: airrohr-firmware/ext_def.h [hunk 2]
```diff
@@ -199,6 +199,8 @@ static const char MEASUREMENT_NAME_INFLUX[] PROGMEM = " ";
 #define PPD_PIN_PM2 GPS_SERIAL_RX
 #endif
 
+//SPH0645 MEMS Microphone
+#define SPHO645_READ  0
```
Comment (chegejames) [inline airrohr-firmware/ext_def.h]: @MCLEANS I believe we want this to default to true

File: airrohr-firmware/SPH0645.h [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+  // Start transmission
+  SLCTXL |= SLCTXLS;
+}
+
+/**
+ * Triggered when SLC has finished writing
+ * to one of the buffers.
+ */
+void ICACHE_RAM_ATTR
```
Comment (chegejames) [inline airrohr-firmware/SPH0645.h]: @MCLEANS let's please stick to one styling guide.

File: airrohr-firmware/ext_def.h (hunk 2 above)
Comment (chegejames) [reply airrohr-firmware/ext_def.h]: @MCLEANS also missing is are the pin assignments

**Review by chegejames — CHANGES_REQUESTED** [review]
@MCLEANS please look at the comments for the changes requested.

**Comment by MCLEANS** [comment]
@chegejames I have made the changes as requested.

**Pull request merged**